scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -947,7 +947,7 @@ def gmres_loose(A, b, tol):
     """
     b = np.asarray(b)
     min_tol = 1000 * np.sqrt(b.size) * np.finfo(b.dtype).eps
-    return gmres(A, b, tol=max(tol, min_tol), atol=0)
+    return gmres(A, b, tol=max(tol, min_tol), debug=1)
```

Comment:
This change looks unintentional.

Reply:
I added the option parameter `debug=1`  because for GMRES (include CG etc), one usually use the calculated result as residual when computing residuals, such as GMRES use the obtained `g[k+1]` rather than `||b - Ax||` (computing `b - Ax` will bring to a (more) matrix-vector multiplication in each iteration). But in the case of singular preconditioner, `g[k+1]` is different from `||b - Ax||`  (In fact, the case of singular preconditioner may not make sense in numerical computing). So I set `debug=1` to computing residuals by `||b - Ax||` rather than `g[k+1]`.

Reply:
Okay, seems like something you want to remove again then.

Reply:
The `atol` will be recovered to keep compatible. I'd like to ask if I can add a new named GMRES function while the original `gmres(...)` still remains

Reply:
> This change looks unintentional.

Indeed, in arpack.py, the option parameter `debug=1` is redundant and I'll remove it (No singular preconditioner is involved in this case), thanks

Reply:
What would the new `GMRES` do differently? A separate function doesn't sound appealing, it'd be better to add a more sanely named keyword to `gmres` if possible.

Reply:
Dear @rgommers,

There is something wrong with inner iteration tolerance in GMRES with preconditioner (unpreconditioned GMRES is OK at present), and I'll try to debug the error. Before that, firstly I would like to make the first PR to submit iterative methods except for GMRES (such as CG, CGS, iCGS, BiCG, BiCGStab, CGNR and CGNE), these algorithms recover all parameter options and are compatible with previous versions (the performance is still better than original codes). And then I'll try to fix the errors in preconditioned GMRES and complete the second PR. Thanks.